modules/latex/latex.js

```diff
@@ -0,0 +1,19 @@
+( function( $ ) {
+	'use strict';
+	$( document.body ).on( 'post-load', function() {
+		// New posts have been added to the page. Re-typeset MathJax if necessary.
+		if ( typeof MathJax !== 'undefined' ) {
+			$( '.infinite-wrap' ).not( '.mathjax-typeset' ).each( function() {
+				var wrap = this;
+
+				MathJax.Hub.Queue( [
+					'Typeset',
+					MathJax.Hub,
+					wrap
+				] );
+
+				$( wrap ).addClass( 'mathjax-typeset' );
+			} );
```

Comment:
This is the most stable version I've been able to achieve so far, but MathJax still sometimes barfs when rendering as a user scrolls up and down.